### DISCUSSION TITLE
Conditional that depends on another conditional

### DISCUSSION THREAD

**Discussion by tangsl** [react-hook-form/react-hook-form#2524] [Q&A]
Hello everyone!

I have a form with some conditional fields, but in certain cases the conditional fields can depend on other conditional fields which causes strange behavior in the fields rendering.

#### Example

A form with 4 fields where 3 are conditional and each depends on the previous field. Enabling the conditionals works perfectly, but when the first field is disabled this is the behavior: 

![conditional](https://user-images.githubusercontent.com/53351154/89704817-54ec2180-d925-11ea-966d-c362079e65e5.gif)

**The question is:** Is this the expected behavior? if so, any advice for removing the other conditionals when the first field is disabled?

Thanks you.

**Code Sandbox**: https://codesandbox.io/s/react-hook-form-conditional-fields-rkbj2?file=/src/index.js

**Comment by bluebill1049** [comment]
are you after `shouldUnregister: false` at `useForm`?

**Reply by bluebill1049** [reply]
honestly, the easiest is probably just toggle the css display property, cheapest and most efficient too.

**Reply by tangsl** [reply]
Thanks  for your answer. I'll give it a try

**Comment by tedliang** [comment]
This is not related to RHF. Solution: https://codesandbox.io/s/react-hook-form-conditional-fields-forked-zoc59
